src/hooks/useAlertPermissions.ts

```diff
@@ -0,0 +1,14 @@
+import { getUserPermissions } from 'data/permissions';
+
+import { useDSPermission } from './useDSPermission';
+
+export function useAlertPermissions() {
+  const { canReadAlerts, canWriteAlerts, canDeleteAlerts } = getUserPermissions();
+  const canEditAlertInDs = useDSPermission(`metrics`, `alert.instances.external:write`);
+
+  return {
+    canReadAlerts,
```

Comment:
I removed Query permissions for `Viewer` in my metrics datasource and I get faced with an infinitely loading spinner. This should be a simple case of checking if there is a value being returned for `useMetricsDS()` and adding it to the return for each of the `canReadAlerts` / `canWriteAlerts` / `canDeleteAlerts`.

![The Alerts page. It shows the Alerts intro text follow by a loading spinner.](https://github.com/user-attachments/assets/dcb416b3-a2b7-4db5-afe8-4d127271760a)